astropy/nddata/utils.py

```diff
@@ -916,3 +925,373 @@ def bbox_cutout(self):
+            try:
+                cutout = cutcls(position, size=(y_pix, x_pix))
+            except NoConvergence:
+                if verbose:
+                    log.info('WCS solution did not converge: '
+                             'Skipping {0}'.format(row['id']))
+                cutouts.append(None)
+                continue
+            except NoOverlapError:
+                if verbose:
+                    log.info('Cutout is not on image: '
+                             'Skipping {0}'.format(row['id']))
+                cutouts.append(None)
+                continue
+            else:
+                cutout_hdr = cutout.wcs.to_header()
+                cutout_arr = cutout.data
+
+        # If cutout result is empty, skip that target
+        if np.array_equiv(cutout_arr, 0):
+            if verbose:
+                log.info('No data in cutout: Skipping {0}'.format(row['id']))
+            cutouts.append(None)
+            continue
+
+        # Finish constructing header.
+        cutout_hdr['OBJ_RA'] = (position.ra.deg, 'Cutout object RA in deg')
+        cutout_hdr['OBJ_DEC'] = (position.dec.deg, 'Cutout object DEC in deg')
+
+        cutouts.append(NDData(data=cutout_arr, wcs=WCS(cutout_hdr), meta=cutout_hdr))
+
+    return cutouts
+
+
+def cutouts_from_fits(image, catalog, image_ext=0, origin=0,
+                      output_dir=None, overwrite=False, verbose=True):
+    """Wrapper for the make_cutouts function. This function will take in a single
+    fits image and return an array containing a list of cutouts as fits hdus. It
+    will also save the cutouts to file if requested.
+
+    Notes
+    -----
+    The input Catalog must have the following columns, which must have
+    `~astropy.unit.Unit`s where applicable:
+
+        * ``'id'`` - ID string; no unit necessary.
+        * ``'coords'`` - SkyCoord (Overrides ra, dec, x and y columns).
+        * ``'ra'`` or ``'x'``- RA (angular units e.g., deg, H:M:S, arcsec etc..)
+          or pixel x position (only in `~astropy.units.pix`).
+        * ``'dec'`` or ``'y'`` - Dec (angular units e.g., deg, D:M:S, arcsec etc..)
+          or pixel y position (only in `~astropy.units.pix`).
+        * ``'cutout_width'`` - Cutout width (e.g., in arcsec, pix).
+        * ``'cutout_height'`` - Cutout height (e.g., in arcsec, pix).
+
+    Optional columns:
+        * ``'cutout_pa'`` - Cutout angle (e.g., in deg, arcsec). This is only
+          use if user chooses to rotate the cutouts. Positive value
+          will result in a clockwise rotation.
+
+    If saved to fits, cutouts are organized as follows:
+        <output_dir>/
+            <id>.fits
+
+    Each cutout image is a simple single-extension FITS with updated WCS.
+    Its header has the following special keywords:
+
+        * ``OBJ_RA`` - RA of the cutout object in degrees.
+        * ``OBJ_DEC`` - DEC of the cutout object in degrees.
+        * ``OBJ_ROT`` - Rotation of cutout object in degrees.
+
+    Examples
```

Comment:
The `Examples` section goes after the `Notes` section (at the bottom).  `Examples` are always last.